### ISSUE TITLE
`supabase link` not updating outdated `storage-api` service

### ISSUE THREAD

**Issue by boozedog** [supabase/cli#2473]
**Describe the bug**
I'm getting this warning
```
You are running outdated service versions locally: 
supabase/storage-api:v1.0.6 => v1.6.4
Run supabase link to update them.
```

But running `supabase link` does not update the outdated `storage-api` service

**Screenshots**
after `supabase link`:
```
❯ supabase services

  
        SERVICE IMAGE      │      LOCAL       │  LINKED
  ─────────────────────────┼──────────────────┼────────────
    supabase/postgres      │ 15.1.1.25        │ 15.1.1.25
    supabase/gotrue        │ v2.152.1         │ v2.152.1
    postgrest/postgrest    │ v12.0.2          │ v12.0.2
    supabase/realtime      │ v2.28.32         │ -
    supabase/storage-api   │ v1.0.6           │ v1.6.4
    supabase/edge-runtime  │ v1.54.6          │ -
    supabase/studio        │ 20240422-5cf8f30 │ -
    supabase/postgres-meta │ v0.80.0          │ -
    supabase/logflare      │ 1.4.0            │ -
    supabase/supavisor     │ 1.1.56           │ -
    darthsim/imgproxy      │ v3.8.0           │ -

Run supabase link to sync your local image versions with the linked project.
```
**System information**
```
❯ cat /etc/lsb-release 
DISTRIB_CODENAME=vicuna
DISTRIB_DESCRIPTION="NixOS 24.11 (Vicuna)"
DISTRIB_ID=nixos
DISTRIB_RELEASE="24.11"
LSB_VERSION="24.11 (Vicuna)"

❯ node --version
v20.11.1

❯ supabase --version
1.178.2

❯ docker --version
Docker version 24.0.9, build v24.0.9
```

**Comment by sweatybridge** [comment]
I can't reproduce this locally. But as an alternative, you can manually update the storage version to match hosted.

```bash
echo "v1.6.4" > supabase/.temp/storage-version
```

**Comment by haydn** [comment]
@sweatybridge FYI, I don't think `supabase link` has ever successfully updated the image versions for me. I just tried again now:

```
> supabase --version
1.204.3

> supabase services


        SERVICE IMAGE      │      LOCAL       │   LINKED
  ─────────────────────────┼──────────────────┼─────────────
    supabase/postgres      │ 14.1.0.89        │ 15.6.1.132
    supabase/gotrue        │ v2.158.1         │ v2.163.0
    postgrest/postgrest    │ v12.2.0          │ v12.2.3
    supabase/realtime      │ v2.30.34         │ -
    supabase/storage-api   │ v1.11.13         │ v1.11.14
    supabase/edge-runtime  │ v1.58.12         │ -
    supabase/studio        │ 20240930-16f2b8e │ -
    supabase/postgres-meta │ v0.83.2          │ -
    supabase/logflare      │ 1.4.0            │ -
    supabase/supavisor     │ 1.1.56           │ -

Run supabase link to sync your local image versions with the linked project.

> supabase link --project-ref «PROJECT REF»
Enter your database password (or leave blank to skip):
Finished supabase link.

> supabase services


        SERVICE IMAGE      │      LOCAL       │   LINKED
  ─────────────────────────┼──────────────────┼─────────────
    supabase/postgres      │ 14.1.0.89        │ 15.6.1.132
    supabase/gotrue        │ v2.158.1         │ v2.163.0
    postgrest/postgrest    │ v12.2.0          │ v12.2.3
    supabase/realtime      │ v2.30.34         │ -
    supabase/storage-api   │ v1.11.13         │ v1.11.14
    supabase/edge-runtime  │ v1.58.12         │ -
    supabase/studio        │ 20240930-16f2b8e │ -
    supabase/postgres-meta │ v0.83.2          │ -
    supabase/logflare      │ 1.4.0            │ -
    supabase/supavisor     │ 1.1.56           │ -

Run supabase link to sync your local image versions with the linked project.
```

**Comment by pewinski** [comment]
I'm having the same problems. `npx supabase link` finishes with no warnings, then `npx supabase start` warns about 

```
WARNING: You are running different service versions locally than your linked project:
supabase/storage-api:v1.22.6 => v1.22.4
Run supabase link to update them.
```

Any updates on this? @sweatybridge suggestion did not worked for me, the `.temp/storage-version` is always overwritten with `custom-metadata` no matter what value I put in there

**Comment by sweatybridge** [comment]
Hi @pewinski, we changed the way storage migrations are pinned, so your storage-api image version will always be upgraded together with cli version. I will make sure the warning message is no longer shown.

**Comment by logemann** [comment]
I have the same error and terminal output as @pewinski . A bit weird that the linked version is a lower version compared to the local installed one right? 

I never understood what the "link" command has to do with upgrades of the local supabase services.

**Comment by sweatybridge** [comment]
> I never understood what the "link" command has to do with upgrades of the local supabase services.

The purpose of link command is to configure your local environment the same way as your remote project. Running different versions could result in undefined behaviour when diff migrations, generating types, etc. That's the primary reason we check that when running supabase link.

> A bit weird that the linked version is a lower version compared to the local installed one right?

Yes, the reason is we do staged rollouts for platform services so some projects will not immediately upgraded to the latest version. On the other hand, local is always updated to track the latest so sometimes it can be ahead.

**Comment by Romerolweb** [comment]
I have linked my project, but I still get 
```bash
npx supabase start
WARNING: You are running different service versions locally than your linked project:
supabase/postgres:15.8.1.093 => supabase/postgres:17.6.1.072
supabase/storage-api:v1.33.5 => v1.33.0
```

**Comment by sai-sy** [comment]
this is not closed I get the same error regardless of how many times I link or delete the .temp folder and relink @sweatybridge

**Comment by engPabloMartinez** [comment]
yeah getting the same as @Romerolweb and @sai-sy.. tried everything... still getting the supabase/storage-api:v1.33.5 => v1.33.0

**Comment by ajstair** [comment]
@sweatybridge we are also still getting the warning – seems like a bug, this issue doesn't appear to be resolved.

### TLDR: 
- The storage-api version warning still appears, contradicting https://github.com/supabase/cli/pull/3488 – this issue should be reopened
- We can't find a documented way to set `storage-version`, and the mismatch causes migration failures locally that blocks local development for our team

### Version Warning
After running `supabase start` in CLI version 2.75.0 (installed via homebrew):
```log
WARNING: You are running different service versions locally than your linked project:
supabase/storage-api:v1.35.3 => v1.33.0
Run supabase link to update them.
```


### Version Mismatch Causes Migration Errors; Can't Update `storage-version`
Furthermore, the version mismatch (and inability to resolve it because we can't configure storage-version manually) is **blocking my devs from their local SDLC.** 

After generating a migration using `supabase db pull`, I get errors with `supabase start` because the migration references functions that don't exist in the old 1.33.0 version (which I can't find any way to change). 
- Running `supabase link` does not update the version. 
- Changing the versions in `supabase/.temp/storage-version` has no effect. 
- Deleting `supabase/.temp` and linking again does not update `storage-version`. 

Full log:
```log
% supabase start           
WARNING: You are running different service versions locally than your linked project:
supabase/storage-api:v1.35.3 => v1.33.0
Run supabase link to update them.
[+] Pulling 13/13
 ✔ vector Skipped - Image is already present locally                                                               0.0s 
 ✔ api Skipped - Image is already present locally                                                                  0.0s 
 ✔ edgeRuntime Skipped - Image is already present locally                                                          0.0s 
 ✔ analytics Skipped - Image is already present locally                                                            0.0s 
 ✔ gateway Skipped - Image is already present locally                                                              0.0s 
 ✔ auth Skipped - Image is already present locally                                                                 0.0s 
 ✔ mailpit Skipped - Image is already present locally                                                              0.0s 
 ✔ pgmeta Skipped - Image is already present locally                                                               0.0s 
 ✔ db Skipped - Image is already present locally                                                                   0.0s 
 ✔ imgProxy Skipped - Image is already present locally                                                             0.0s 
 ✔ studio Skipped - Image is already present locally                                                               0.0s 
 ✔ realtime Skipped - Image is already present locally                                                             0.0s 
 ✔ storage Skipped - Image is already present locally                                                              0.0s 
Starting database...
Initialising schema...
Seeding globals from roles.sql...
Applying migration 20260204034905_remote_schema.sql...
Stopping containers...
ERROR: function storage.delete_prefix_hierarchy_trigger() does not exist (SQLSTATE 42883)                                                          
At statement: 1131                                                                                                                                 
CREATE TRIGGER objects_delete_delete_prefix AFTER DELETE ON storage.objects FOR EACH ROW EXECUTE FUNCTION storage.delete_prefix_hierarchy_trigger()
Try rerunning the command with --debug to troubleshoot the error.
```

The offending SQL statement in the generated xxxxxxxxxxxxxxxxx_remote_schema.sql
```SQL
CREATE TRIGGER objects_delete_delete_prefix AFTER DELETE ON storage.objects FOR EACH ROW EXECUTE FUNCTION storage.delete_prefix_hierarchy_trigger();
-- other storage schema statements have similar issues
```


It seems like maybe the `storage-version` version is ignored (and determined by something else e.g. `storage-migration`)?

**Comment by YawenD** [comment]
Same issue here. Downgrading to v2.72.8 solved the issue.

**Comment by lenardflx** [comment]
+1. Downgrading did not fix it for me.

**Comment by conte91** [comment]
We're also very much affected, this has made it impossible to pull remote schemas :( (#4815)

**Comment by coachkarzan** [comment]
After updating to 2.75, I ran into this problem. Downgrading didn't help either.

**Comment by giorgiogross** [comment]
Same here

**Comment by hoverbird** [comment]
Same issue, downgrading did not fix this for me. This is really blocking my ability to develop against sup abase....
